sdk/src/client/secret/ledger_nano.rs

```diff
@@ -316,36 +316,32 @@ impl SecretManage for LedgerSecretManager {
 
             let mut remainder_index = 0u16;
             if let Some(remainder_address) = remainder_address {
-                match &prepared_transaction.essence {
-                    TransactionEssence::Regular(essence) => {
-                        // find the index of the remainder in the essence
-                        // this has to be done because outputs in essences are sorted
-                        // lexically and therefore the remainder is not always the last output.
-                        // The index within the essence and the bip32 index will be validated
-                        // by the hardware wallet.
-                        // The outputs in the essence already are sorted
-                        // at this place, so we can rely on their order and don't have to sort it again.
-                        'essence_outputs: for output in essence.outputs().iter() {
-                            if let Output::Basic(s) = output {
-                                if let Some(address) = s.unlock_conditions().address() {
-                                    if *remainder_address == *address.address() {
-                                        break 'essence_outputs;
-                                    }
-                                }
-                            } else {
-                                log::debug!("[LEDGER] unsupported output");
-                                return Err(Error::MiscError.into());
+                // find the index of the remainder in the essence
+                // this has to be done because outputs in essences are sorted
+                // lexically and therefore the remainder is not always the last output.
+                // The index within the essence and the bip32 index will be validated
+                // by the hardware wallet.
+                // The outputs in the essence already are sorted
+                // at this place, so we can rely on their order and don't have to sort it again.
```

Comment:
Some wrong comments here, outputs don't need to be sorted anymore

Reply:
Changed